config/cmake/HDFCXXCompilerFlags.cmake

```diff
@@ -115,106 +111,24 @@ if (NOT MSVC AND NOT MINGW)
     endif ()
   endif ()
 
-  #-----------------------------------------------------------------------------
-  # Option to allow the user to enable developer warnings
-  # Developer warnings (suggestions from gcc, not code problems)
-  #-----------------------------------------------------------------------------
-  if (HDF5_ENABLE_DEV_WARNINGS)
-    message (STATUS "....HDF5 developer group warnings are enabled")
-  #  if (CMAKE_CXX_COMPILER_ID STREQUAL "Intel")
-  #    list (APPEND H5_CXXFLAGS "-Winline -Wreorder -Wport -Wstrict-aliasing")
-  #  elseif (CMAKE_CXX_COMPILER_ID STREQUAL "GNU")
-    if (CMAKE_CXX_COMPILER_ID STREQUAL "GNU")
-      # autotools always add the C flags with the CXX flags
-      ADD_H5_FLAGS (H5_CXXFLAGS "${HDF5_SOURCE_DIR}/config/gnu-warnings/developer-general")
-  #  elseif (CMAKE_CXX_COMPILER_ID STREQUAL "Clang")
-  #    ADD_H5_FLAGS (H5_CXXFLAGS "${HDF5_SOURCE_DIR}/config/clang-warnings/developer-general")
-    endif ()
-  else ()
-    if (CMAKE_CXX_COMPILER_ID STREQUAL "GNU")
-      # autotools always add the C flags with the CXX flags
-      ADD_H5_FLAGS (H5_CXXFLAGS "${HDF5_SOURCE_DIR}/config/gnu-warnings/no-developer-general")
-  #  elseif (CMAKE_CXX_COMPILER_ID STREQUAL "Clang")
-  #    ADD_H5_FLAGS (H5_CXXFLAGS "${HDF5_SOURCE_DIR}/config/clang-warnings/no-developer-general")
-    endif ()
-  endif ()
-
   if (CMAKE_CXX_COMPILER_ID STREQUAL "GNU")
-    # Technically, variable-length arrays are part of the C99 standard, but
-    #   we should approach them a bit cautiously... Only needed for gcc 4.X
-    if (CMAKE_CXX_COMPILER_VERSION VERSION_LESS 5.0 AND CMAKE_C_COMPILER_VERSION VERSION_GREATER_EQUAL 4.8)
-      # autotools always add the C flags with the CXX flags
-      ADD_H5_FLAGS (H5_CXXFLAGS "${HDF5_SOURCE_DIR}/config/gnu-warnings/4.8-4.last")
-    endif ()
-
-    # Append more extra warning flags that only gcc 4.8+ know about
-    if (NOT CMAKE_CXX_COMPILER_VERSION VERSION_LESS 4.8)
-      ADD_H5_FLAGS (H5_CXXFLAGS "${HDF5_SOURCE_DIR}/config/gnu-warnings/4.8")
-      if (HDF5_ENABLE_DEV_WARNINGS)
-        ADD_H5_FLAGS (H5_CXXFLAGS "${HDF5_SOURCE_DIR}/config/gnu-warnings/developer-4.8")
-      else ()
-        ADD_H5_FLAGS (H5_CXXFLAGS "${HDF5_SOURCE_DIR}/config/gnu-warnings/no-developer-4.8")
-      endif ()
-    endif ()
-
-    # Append more extra warning flags that only gcc 4.9+ know about
+    # Append more extra warning flags that only gcc 4.9+ knows about
     if (NOT CMAKE_CXX_COMPILER_VERSION VERSION_LESS 4.9)
-      # autotools always add the C flags with the CXX flags
-      ADD_H5_FLAGS (H5_CXXFLAGS "${HDF5_SOURCE_DIR}/config/gnu-warnings/4.9")
```

Comment:
This is (a) not true and (b) triggers an enormous number of "not a C++ warning" warnings.